### PR TITLE
catch all, with better format

### PR DESCRIPTION
**Pull request by zhreshold**
## Description ##
Improve the tools/diagnose.py script

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Improve catch exception and feedback.

### CODE REVIEW

File: tools/diagnose.py [hunk 1]
```diff
@@ -81,76 +81,98 @@ def test_connection(name, url, timeout=10):
...
+            ch = f.read().strip()
+            print('Commit Hash   :', ch)
+    except ImportError:
+        print('No MXNet installed.')
+    except IOError:
+        pass
+    except:
+        print("An error occured trying to import mxnet.")
+        print("This is very likely due to a missing dll.")
+        raise
```
Comment (szha) [inline tools/diagnose.py]: Is there a way to print the error? Sometimes there is useful information such as missing library file or missing symbol.

File: tools/diagnose.py [hunk 2]
```diff
@@ -81,76 +81,98 @@ def test_connection(name, url, timeout=10):
...
-            print('Directory    :', mx_dir)
-            commit_hash = os.path.join(mx_dir, 'COMMIT_HASH')
-            with open(commit_hash, 'r') as f:
-                ch = f.read().strip()
-                print('Commit Hash   :', ch)
-        except ImportError:
-            print('No MXNet installed.')
-        except IOError:
-            pass
+        check_mxnet()
```
Comment (szha) [inline tools/diagnose.py]: Maybe do mxnet last.

**Comment by zhreshold** [comment]
How about this? @szha

File: tools/diagnose.py [hunk 3]
```diff
@@ -110,12 +110,12 @@ def check_mxnet():
...
-        pass
-    except:
-        print("An error occured trying to import mxnet.")
-        print("This is very likely due to a missing dll.")
-        raise
+    except Exception as e:
+        import traceback
+        if not isinstance(e, IOError):
+            print("An error occured trying to import mxnet.")
+            print("This is very likely due to missing dll files.")
```
Comment (szha) [inline tools/diagnose.py]: missing or incompatible library files (just making sure the script doesn't assume windows users)

**Pull request merged**